### ISSUE TITLE
Rebase upstream

### ISSUE THREAD

**Issue by chrert** [Russell91/apollocaffe#42]
Are there any plans to keep the fork up to date with upstream caffe? I'm afraid there have been a lot of structural changes in the upstream repo, but it would be nice to be in sync with it...

**Comment by Russell91** [comment]
Yeah, this is an interesting idea. I think if we were to do an update one thing to consider would be ditching the ApolloCaffe layer api for the net_spec API, just to be more compatible with Caffe. The "dynamic network" capabilities are an intriguing but most likely unused feature. I think people are more interested in just having memory and compute efficient RNNs and such. The biggest question for me is, what are people using ApolloCaffe for that they can't get in Caffe, and is there a way to just focus on those features in a new version.
